src/shims/unix/android/foreign_items.rs

```diff
@@ -25,8 +26,65 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 this.write_scalar(errno_place.to_ref(this).to_scalar(), dest)?;
             }
 
+            // Threading
+            "prctl" => {
+                // We do not use `check_shim` here because `prctl` is variadic. The argument
+                // count is checked bellow.
+                this.check_abi_and_shim_symbol_clash(abi, Abi::C { unwind: false }, link_name)?;
+
+                check_args_len("prctl", args, 1)?;
+
+                let id = this.read_scalar(&args[0])?.to_i32()?;
+                // FIXME: Use PR_SET_NAME and PR_GET_NAME constants when
+                // https://github.com/rust-lang/libc/pull/3941 lands.
+                const PR_SET_NAME: i32 = 15;
+                const PR_GET_NAME: i32 = 16;
+
+                let res = match id {
+                    PR_SET_NAME => {
+                        check_args_len("'PR_SET_NAME' prctl", args, 2)?;
```

Comment:
```suggestion
                        check_args_len("prctl(PR_SET_NAME, ...)", args, 2)?;
```

Reply:
It contradicts with another suggestion if `check_args_len` made even with `check_arg_count` as you asked. Should then `check_min_arg_count` accept a function name as the first argument?

Reply:
Yeah it seems to me it would make sense to pass the function name to `check_min_arg_count`. (`check_arg_count` doesn't take a function name and we should probably fix that.)